### PR TITLE
Adding incoming and outgoing indicators to Mercurial prompt. 

### PR DESCRIPTION
**Pull request by Siecje**
Requires the third party Mercurial extension [hg prompt](https://bitbucket.org/sjl/hg-prompt/src) to be installed.

Issue: https://github.com/fish-shell/fish-shell/issues/2858

@zanchey on IRC suggested using a cache file and invalidating it after an ```hg push```, ```hg pull``` etc with

```fish
function hg;
    if contains pull "$argv"
        # *invalidate cache*
    end
    command hg $argv
end
```

### CODE REVIEW

File: share/functions/__fish_hg_prompt.fish [hunk 1]
```diff
@@ -50,17 +53,32 @@ function __fish_hg_prompt --description 'Write out the hg prompt'
         set branch "$branch|$bookmark"
     end
 
+    # Only add incoming and outgoing if hg prompt is installed
+    if command -s "hg prompt" > /dev/null
```
Comment (Siecje) [inline share/functions/__fish_hg_prompt.fish]: This check if the ```hg prompt``` command exists, doesn't work, it will always be false.

**Comment by krader1961** [comment]
> This check if the hg prompt command exists, doesn't work, it will always be false.

Correct, because `hg` is a command but `hg prompt` is not, just as `echo` is a command but `echo prompt` is not. Are you asking for help in resolving that problem or simply pointing out you're aware of it and a fix is forthcoming?

**Comment by faho** [comment]
I don't quite like the idea of using a third-party extension for this, but it seems `hg incoming` and `hg outgoing` will always go over the network - which is just prohibitively slow (in the best case it takes about a second).

However, I can't seem to get the prompt extension to work. I've cloned an hg repo, made and committed a change so `hg outgoing` shows something, but `hg prompt '{outgoing|count}'` just happily exits without output and status 0.

Checking if the extension exists should be a matter of checking the hgrc file for a "prompt=" line (with optional whitespace), and if it exists we might as well use it for everything.

**Comment by faho** [comment]
Also, the extension seems to be kind of dead - the last commit was 8 months ago and there's an open issue about it not working with hg 2.9 - I now have hg 3.7.2.

**Comment by Siecje** [comment]
@faho To install hg prompt you need to clone the repo and then add 

```prompt= /path/to/repo/prompt.py```

Under the ```[extensions]``` in .hgrc

**Comment by faho** [comment]
@Siecje: Yes, I know, I've done that - otherwise I'd probably get an error message. `hg prompt '{branch}'` also works, just the incoming/outgoing stuff doesn't (I haven't checked much else), and I'm assuming it's a version incompatibility. One which would make this approach pretty much worthless.

**Comment by faho** [comment]
I'm sorry, this approach doesn't seem to work. If anyone figures out how to make it work, we'll reopen.

**Pull request closed without merging**